### PR TITLE
Fix forward AD for `torch.angle`

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77269
* __->__ #77267

Fixes: https://github.com/pytorch/pytorch/issues/77246

### CODE REVIEW

**Review by albanD — APPROVED** [review]

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -260,7 +260,7 @@
 
 - name: angle(Tensor self) -> Tensor
   self: angle_backward(grad, self)
-  result: handle_r_to_c(result.scalar_type(), angle_backward(self_t, self_p))
+  result: handle_r_to_c(result.scalar_type(), angle_backward(self_t.conj(), self_p).conj())
```
Comment (Lezcano) [inline tools/autograd/derivatives.yaml:263]: fwiw, the `r_to_c` returns a real tensor, so the final `.conj()` is not needed.

Comment (soulitzer) [reply tools/autograd/derivatives.yaml:263]: ah good catch!

**Pull request closed without merging**